bucket.go

```diff
@@ -21,41 +21,46 @@ const (
 // insert a fingerprint into a bucket. Returns true if there was enough space and insertion succeeded.
 // Note it allows inserting the same fingerprint multiple times.
 func (b *bucket) insert(fp fingerprint) bool {
-	for i, tfp := range b {
-		if tfp == nullFp {
-			b[i] = fp
-			return true
-		}
+	if i := b.index(nullFp); i != 4 {
+		b[i] = fp
+		return true
 	}
 	return false
 }
 
 // delete a fingerprint from a bucket.
 // Returns true if the fingerprint was present and successfully removed.
 func (b *bucket) delete(fp fingerprint) bool {
-	for i, tfp := range b {
-		if tfp == fp {
-			b[i] = nullFp
-			return true
-		}
+	if i := b.index(fp); i != 4 {
+		b[i] = nullFp
+		return true
 	}
 	return false
 }
 
 func (b *bucket) contains(needle fingerprint) bool {
-	for _, fp := range b {
-		if fp == needle {
-			return true
-		}
+	return b.index(needle) != 4
+}
+
+func (b *bucket) index(needle fingerprint) uint8 {
```

Comment:
this loop unrolling assumes that the bucketSize is guaranteed to be 4, so that constant should be removed if the intent is to unroll it.

Reply:
`bucketSize` is used in several places in cuckoofilter.go. maybe we could add a test for `len(bucket{}) == bucketSize`?

Reply:
nvm i thought of a better way to do this.